Домашнее задание

Ulyanov-VN opened the pull request.

Ulyanov-VN committed:
Домашнее задание

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/SettingsActivity.kt:
```diff
@@ -0,0 +1,23 @@
+
+        button4.setOnClickListener {
+            val displayIntent = Intent(this, MainActivity::class.java)
+            startActivity(displayIntent)
```
Для выхода с экрана стоит использовать функцию `finish()`, а не `startActivity`, что создаст новую активити и откроет её. Принципиальная разница заключается в том, что в твоём коде вместо возврата открывается и создаётся новый экран с новым состоянием, а при использовании `finish()` закрывается текущий экран и показывается предыдущий, который был открыт перед текущим.

ArturNurtdinov commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -3,17 +3,111 @@
+
+                />
+        </RelativeLayout>
+    </RelativeLayout>
```
Вложенный RelativeLayout здесь лишний

ArturNurtdinov commented on app/src/main/res/layout/activity_media.xml:
```diff
@@ -0,0 +1,16 @@
+android:layout_height="wrap_content"
+android:layout_gravity="center"
+android:gravity="center"
+android:text="Я сделал Медиатеку!"/>
```
Тут выравнивание чуть съехало, в Android студии можно сделать авто форматирование, нажав ПКМ по нужному классу и выбрав `Reformat code`, или нажать сочетание клавиш из соответствующего пункта в меню.

ArturNurtdinov commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,183 @@
+        android:id="@+id/line_share_app"
+        android:layout_width="match_parent"
+        android:layout_height="0dp"
+        android:layout_marginTop="141dp"
```
Не стоит все элементы через отступ сверху распологать, т.к. если пользователь увеличит шрифт, то текст наедет друг на друга, лучше сделать LinearLayout, чтобы текст располагался друг за другом с отступом между текстом

Ulyanov-VN committed:
Домашнее задание с учетом замечаний

Ulyanov-VN committed:
Merge pull request #2 from Ulyanov-VN/dz_s8_wc1

Домашнее задание с учетом замечаний

Ulyanov-VN committed:
Домашнее задание Модуль №9

Ulyanov-VN committed:
Домашнее задание Модуль №9

Ulyanov-VN committed:
Merge remote-tracking branch 'origin/dz_m9' into dz_m9

Ulyanov-VN committed:
Домашнее задание Модуль №9 (Исправление замечаний)
1. Исправил все крически важные замечания (учитывая что нужно было убрать панель навигации, внес изменения в SearchActivity.kt)
2. Прошу прощения, что когда много закоменчено, к следующей домашней работе я удалю все. Часто пишу на работе, мобильный интернет вообще не работает, а на рабочем компьютеры все ресурсы заблокированы
3. Все рекомендации по коду буду исправлять по ходу выполнения следующих домашних заданий, спасибо за комментарии.

Ulyanov-VN committed:
Merge pull request #3 from Ulyanov-VN/dz_m9

Домашнее задание Модуль №9

Ulyanov-VN committed:
Домашнее задание Модуль №10

Ulyanov-VN committed:
Домашнее задание Модуль №10 - устранение замечаний

1. Плейсхолдер обложки сделаны по макету для темной темы и светлой темы. Так же дабавлены строчки по очистке кеша, чтобы при отключении интернета отображались drawable.placeholder_dark/drawable.placeholder_light

2. Исправлен отступ между текстом и обложкой на 8dp.

Ulyanov-VN committed:
Merge pull request #4 from Ulyanov-VN/dz_m10

Домашнее задание Модуль №10

Ulyanov-VN committed:
Домашнее задание Модуль №11

Ulyanov-VN committed:
Домашнее задание Модуль №11 - устранение замечаний.

1. Исправлен прогресс бар
2. При загрузке треков появляются плейсхолдеры, больше нет белых квадратов
3. Очистка запроса исправлена, список треков удаляется

Анастасия для выполнения домашнего задания, недостаточно информации, которую дает яндекс практикум в своей учебной программе и поэтому 90 процентов информации приходится искать самостоятельно.

Ulyanov-VN committed:
Merge pull request #5 from Ulyanov-VN/dz_m11

Домашнее задание Модуль №11

Ulyanov-VN committed:
Домашнее задание Модуль №12

Ulyanov-VN committed:
Домашнее задание Модуль №12 (устранение замечаний)

Ulyanov-VN committed:
Merge pull request #6 from Ulyanov-VN/dz_m12

Домашнее задание Модуль №12

Ulyanov-VN committed:
Домашнее задание Модуль №13

Ulyanov-VN committed:
Домашнее задание Модуль №13 (устранение замечаний)

Ulyanov-VN committed:
Домашнее задание Модуль №13 (повтороное устранение замечаний)

1. Исправил "Отступ у кнопки "Назад" сверху экрана и слева должен быть 16dp (сейчас 24)": 1. Обратился к наставнику за советом он мне предложил сделать margin = "4dp", а padding="12dp".  2. добавил android:fitsSystemWindows="true" чтобы стрелка не налезала на тулбар. 3. Добавил инокнку 24x24 и сделал корректировку с учетом этой иконки

2. Исправил "Горизонтальный отступ у текстовых полей под временем воспроизведения должен быть 16dp, сейчас - 24dp " строке длительность добавлен android:layout_marginStart="16dp" и android:layout_marginEnd="16dp", а все остальные строки наследуют этот параметр.

Все остальные твои коментарии я сохранил и буду разбираться. Спасибо большое.

Ulyanov-VN committed:
Merge pull request #7 from Ulyanov-VN/dz_m13

Домашнее задание Модуль №13

Ulyanov-VN committed:
Домашнее задание Модуль №14

Ulyanov-VN committed:
Merge pull request #8 from Ulyanov-VN/dz_m14

Домашнее задание Модуль №14

Ulyanov-VN committed:
Домашнее задание Модуль №15

Ulyanov-VN committed:
Домашнее задание Модуль №15 (устранение замечаний)
1.  Приложение работает корректно: возможности, предоставляемые приложением до рефакторинга, предоставляются в том же объёме и качестве - Отступ длительности трека не соответствует макетам в списке треков

Замечание устранено, приведено в соответствие с макетом

2.  Доступ к данным должен осуществляться через репозитории, которые инкапсулируют логику получения и сохранения данных - Доступ к значению темы приложения в SharedPreferences осуществляется через класс ThemeHelper, нужно реализовать репозиторий для хранения данных

Замечания устранил и отработал комментарии выданные в гите

3. Работа с интеракторами организована через интерфейсы - Отсутствуют интерфейсы для FormatTimeInteractor, GetCountryNameInteractor, GetCoverArtworkInteractor, GetReleaseYearInteractor,

Добавлены интерфейсы

Ulyanov-VN committed:
Merge pull request #9 from Ulyanov-VN/dz_m15

Домашнее задание Модуль №15

Ulyanov-VN committed:
Домашнее задание Модуль №16

Ulyanov-VN committed:
Домашнее задание Модуль №16 (устранение замечаний)

1.❌ Ссылка на курс по Андроид-разработке находится в strings.xml и передаётся в Intent через метод контекста getString() - Ссылка на курс отсутствует, вместо ссылки захардкожен текст в SharingInteractorImpl
  ❌ Текст для каждого из предзаполненных полей электронного письма находятся в отдельных строках strings.xml и передаётся в Intent через метод контекста getString() - Все тексты захардкожены в SharingInteractorImpl
  ❌ Ссылка на оферту находится в strings.xml и передаётся в Intent через метод контекста getString() - Ссылка захардкожена в SharingInteractorImpl

  Исправлено.

2. ❌ Все взаимодействия со слоями Model должны осуществляться через ViewModel - SettingsActivity напрямую работает с SharingInteractor

  Исправлено.

Ulyanov-VN committed:
Домашнее задание Модуль №16 (устранение замечаний)

❌ Классы в пакете domain не используют классы из Android SDK - SharingInteractorImpl зависит от Context

Привет, исправил ошибку, класс SharingInteractorImpl больше не зависит от Android SDK (Context).

Ulyanov-VN committed:
Merge pull request #10 from Ulyanov-VN/dz_m16

Домашнее задание Модуль №16

Ulyanov-VN committed:
Домашнее задание Модуль №17

Ulyanov-VN committed:
Merge pull request #11 from Ulyanov-VN/dz_m17

Домашнее задание Модуль №17

Ulyanov-VN committed:
Домашнее задание Модуль №18

Ulyanov-VN committed:
Домашнее задание Модуль №18 (исправление замечаний)

1. Плейсхолдеры на новых экранах не соответствуют макетам (вертикальные отступы)

Сделал вертикальные отпуступы в соответствии с макетом.
Пришлось создать layout-land, чтобы при перевороте экрана все элепенты помещались на экране.

Ulyanov-VN committed:
Merge pull request #12 from Ulyanov-VN/dz_m18

Домашнее задание Модуль №18

Ulyanov-VN committed:
Домашнее задание Модуль №19

Ulyanov-VN committed:
Домашнее задание Модуль №19 (исправление замечаний)

Привет, исправил все отступы и сделал как по макету.

Ulyanov-VN committed:
Merge pull request #13 from Ulyanov-VN/dz_m19

Домашнее задание Модуль №19

Ulyanov-VN committed:
Домашнее задание Модуль №20

Ulyanov-VN committed:
Merge pull request #14 from Ulyanov-VN/dz_m20

Домашнее задание Модуль №20

Ulyanov-VN committed:
Домашнее задание Модуль №21

Ulyanov-VN committed:
Домашнее задание Модуль №21 (исправление замечаний)
Исправил по твоим рекомендациям

Ulyanov-VN committed:
Merge pull request #15 from Ulyanov-VN/dz_m21

Домашнее задание Модуль №21

Ulyanov-VN committed:
Домашнее задание Модуль №22

Ulyanov-VN committed:
Домашнее задание Модуль №22

Ulyanov-VN committed:
Merge pull request #16 from Ulyanov-VN/dz_m22

Dz m22

Ulyanov-VN committed:
Домашнее задание Модуль №23

Ulyanov-VN committed:
Домашнее задание Модуль №23 (исправление замечаний)

Ulyanov-VN committed:
Домашнее задание Модуль №23 (исправление замечаний)

Ulyanov-VN committed:
Домашнее задание Модуль №23 (исправление замечаний)

Ulyanov-VN committed:
Merge pull request #17 from Ulyanov-VN/dz_m23

Домашнее задание Модуль №23

Ulyanov-VN committed:
Домашнее задание Модуль №24

Ulyanov-VN committed:
Домашнее задание Модуль №24 (исправление замечаний)

Ulyanov-VN committed:
Merge pull request #18 from Ulyanov-VN/dz_m24

Домашнее задание Модуль №24
